app/controllers/admin/products_v3_controller.rb

```diff
@@ -176,6 +178,7 @@ def product_query_includes
           :stock_items,
           :tax_category,
           :supplier,
+          :taggings,
```

Comment:
It would be nice to make the include conditional as well to reduce the complexity of when not searching for tags. Or do we need them anyway?

Reply:
We display tags on the page so it should be needed, I included it on [bullet](https://github.com/flyerhzm/bullet) recommendation. It seems to be preloading the tags as expected, but there is still a tag related query per variant, I was planning to double check that as I work on adding the tag rules.